packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,141 @@
+    };
+
+    if (previousParticipants.length > remoteParticipants.length) {
+      const whoLeft = previousParticipants.filter((p) => !remoteParticipants.includes(p));
+      //someone left
+      setParticipantEventString(createAnnouncmentString('left', whoLeft, strings));
+      setPreviousParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > previousParticipants.length) {
+      const whoJoined = remoteParticipants.filter((p) => !previousParticipants.includes(p));
+      // someone joined
+      setParticipantEventString(createAnnouncmentString('joined', whoJoined, strings));
+      setPreviousParticipants(remoteParticipants);
+    }
+  }, [remoteParticipants, previousParticipants, strings]);
+
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  switch (participants.length) {
+    case 1:
+      const oneName = participants[0].displayName ?? strings.unnamedParticipantChangedString;
+      return _formatString(
+        direction === 'joined' ? strings.participantJoinedNoticeString : strings.participantLeftNoticeString,
+        { displayNames: oneName }
+      );
+    case 2:
+      const twoNames = participants.map((p) => p.displayName ?? strings.unnamedParticipantChangedString);
+      return _formatString(
+        direction === 'joined' ? strings.twoParticipantJoinedNoticeString : strings.twoParticipantLeftNoticeString,
+        {
+          displayName1: twoNames[0],
+          displayName2: twoNames[1]
+        }
+      );
+    case 3:
+      const threeNames = participants.map((p) => p.displayName ?? strings.unnamedParticipantChangedString);
+      return _formatString(
+        direction === 'joined' ? strings.threeParticipantJoinedNoticeString : strings.threeParticipantLeftNoticeString,
+        {
+          displayName1: threeNames[0],
+          displayName2: threeNames[1],
+          displayName3: threeNames[1]
+        }
+      );
+    default:
+      break;
+  }
+  /**
+   * We don't want to announce every name when more than 3 participants join at once so
+   * we parse out the first 3 names we have and announce those.
+   */
+  const names =
+    participants.filter((p) => p.displayName).length > 0
+      ? participants
+          .filter((p) => p.displayName)
+          .slice(0, 2)
+          .map((p) => p.displayName ?? strings.unnamedParticipantChangedString)
+          .join(', ')
+      : /** if we have no participants with displayNames we just announce one unnamed participant */
+        strings.unnamedParticipantChangedString;
+
+  const numberOfExtraParticipants =
+    names === strings.unnamedParticipantChangedString ? participants.length - 1 : participants.length - 1;
+
+  const namesPlusExtra =
+    names +
```

Comment:
this will be _really_ hard for contoso (and us) to localize correctly, are we sure that other languages the grammer will work?

Reply:
I would encourage the `participantsOverflowString` to be:
```
"participantsOverflowString": "{displayName1}, {displayName2}, {displayName3}, and {numOfParticipants} other participants have joined"
```

(and then have one for joined and left, and update the name from participantsOverflowString to something like `manyParticipantsJoinedString` per other comment)